lib/livebook/application.ex

```diff
@@ -223,20 +226,20 @@ defmodule Livebook.Application do
     end
   end
 
-  defp create_teams_hub() do
+  defp parse_teams_hub() do
     teams_key = System.get_env("LIVEBOOK_TEAMS_KEY")
     auth = System.get_env("LIVEBOOK_TEAMS_AUTH")
 
     cond do
       teams_key && auth ->
-        Application.put_env(:livebook, :teams_auth?, true)
-
         case String.split(auth, ":") do
           ["offline", name, public_key] ->
-            create_offline_hub(teams_key, name, public_key)
+            Application.put_env(:livebook, :teams_auth, :offline)
+            fn -> create_offline_hub(teams_key, name, public_key) end
 
           ["online", name, org_id, org_key_id, agent_key] ->
-            create_online_hub(teams_key, name, org_id, org_key_id, agent_key)
+            Application.put_env(:livebook, :teams_auth, :online)
```

Comment:
@aleDsz the idea is that you can now configure the identity provider here if none is set.